### PR TITLE
Erlang improvements

### PR DESCRIPTION
**Pull request by ericbmerritt**

### CODE REVIEW

File: doc/erlang-users-guide.xml [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  </para>
+</section>
+<section xml:id="packaging-erlang-applications">
+  <title>Packaging Erlang Applications</title>
+  <section xml:id="rebar3-packages">
+    <title>Rebar3 Packages</title>
+    <para>
+      There is a Nix functional called
+      <literal>buildRebar3</literal>. We use this function to make a
+      derivation that understands how to build the rebar3 project. we
```
Comment (gleber) [inline doc/erlang-users-guide.xml]: s/we/We/

File: doc/erlang-users-guide.xml [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      can the expression for building <literal>hex2nix</literal> in
```
Comment (gleber) [inline doc/erlang-users-guide.xml]: Something's wrong with this sentence

File: doc/erlang-users-guide.xml [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+      };
+  drv = erlangPackages.callPackage f {};
+
+in
+ drv
+    </programlisting>
+    <section xml:id="building-in-a-shell">
+    <title>Building in a shell</title>
+    <para>
+      Unfortunatly forr us users of Nix, Rebar isn't very cooperative
```
Comment (gleber) [inline doc/erlang-users-guide.xml]: s/forr/for/

File: doc/erlang-users-guide.xml [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        nix-shell --run "erl"
+
+shell:
+        nix-shell --run "bash"
+
+bootstrap:
+        nix-shell --pure --run "rebar3-nix-bootstrap"
+
+build: bootstrap
+        nix-shell --pure --run "HOME=$(CURDIR) rebar3 escriptize"
```
Comment (gleber) [inline doc/erlang-users-guide.xml]: This is an example of a Makefile which simplifies work with rebar3 package, which requires doing rebar3-nix-bootstrap once in a while. I do not think that most people do "escriptize" on most of their projects. It'll be a bit better to replace "escriptize" with "compile" here and below.

File: doc/erlang-users-guide.xml [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+      user rebar as follows things should simply work.
+    </para>
+  </section>
+</section>
+</section>
+<section xml:id="generating-packages-from-hex-with-hex2nix">
+  <title>Generating Packages from Hex with Hex2Nix</title>
+  <para>
+    Updating the Hex packages requires the use of the
+    <literal>hex2nix</literal> tool. Given the path to the Erlang
```
Comment (gleber) [inline doc/erlang-users-guide.xml]: I think it's good to provide a link to hex2nix project (to it's README) and mention that it's installable from nixpkgs

**Comment by gleber** [comment]
Some comments on docs. Feel free to merge after fixing worthwhile comments.

File: doc/erlang-users-guide.xml (hunk 4 above)
Comment (ericbmerritt) [inline doc/erlang-users-guide.xml]: Agreed, strongly.

**Comment by ericbmerritt** [comment]
@gleber updated

File: pkgs/development/tools/build-managers/rebar3/registrySnapshot.nix [hunk 6]
```diff
@@ -1,8 +1,23 @@
...
-  repo = "hex-pm-registry-snapshots";
-  rev = "329ae2b";
-  sha256 = "1rs3z8psfvy10mzlfvkdzbflgikcnq08r38kfi0f8p5wvi8f8hmh";
+stdenv.mkDerivation rec {
+    name = "hex-registry";
+    rev = "329ae2b";
+    version = "0.0.0+build.${rev}";
+
+    src = fetchFromGitHub {
+        owner = "gleber";
```
Comment (gleber) [inline pkgs/development/tools/build-managers/rebar3/registrySnapshot.nix]: ah, this can be changed to erlang-nix

**Comment by gleber** [comment]
Thanks for fixes! I think this can now be PR-ed into nixpkgs upstream

**Pull request merged**